### ISSUE TITLE
UX issues in multiple choice and number tasks

### ISSUE THREAD

**Issue by andreia-ferreira** [google/ground-android#3746]
A couple of small issues were found by @jo-spek when testing the latest version:
- Multiple choice task: the entire row should be clickable in order to improve user experience and accessibility
- Number task: typing a non-numeric character clears the entire field
